plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/data/order/OrderingUtils.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * DBeaver - Universal Database Manager
+ * Copyright (C) 2010-2025 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jkiss.dbeaver.model.data.order;
+
+import org.jkiss.code.NotNull;
+import org.jkiss.code.Nullable;
+import org.jkiss.dbeaver.model.DBUtils;
+import org.jkiss.dbeaver.model.data.DBDAttributeBinding;
+import org.jkiss.dbeaver.model.data.DBDAttributeConstraint;
+import org.jkiss.dbeaver.model.data.DBDDataFilter;
+import org.jkiss.dbeaver.model.data.DBDRowIdentifier;
+import org.jkiss.dbeaver.model.preferences.DBPPreferenceStore;
+import org.jkiss.dbeaver.model.runtime.VoidProgressMonitor;
+import org.jkiss.dbeaver.model.struct.DBSDataContainer;
+import org.jkiss.dbeaver.model.struct.DBSEntity;
+import org.jkiss.dbeaver.model.struct.DBSEntityAttribute;
+
+import java.util.List;
+
+/**
+ * Utilities for applying default ordering in a non-UI layer.
+ */
+public final class OrderingUtils {
+
+    private OrderingUtils() {
+        // no instances
+    }
+
+    public static boolean addOrderingOnClientSide(
+        @Nullable DBSDataContainer dataContainer,
+        @NotNull DBDDataFilter dataFilter,
+        @Nullable DBDRowIdentifier rowIdentifier,
+        @NotNull DBPPreferenceStore preferences
```

Comment:
I think you should provide an overload that accepts `OrderingPolicy`, avoiding even touching a preference store. Same for the other method.